compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -170,8 +175,13 @@ void instrumentPackage(BIRPackage pkg) {
                     }
                 }
                 if (serviceName == null) {
-                    serviceName = pkg.packageID.orgName.value + "_" + pkg.packageID.name.value + "_svc_" +
-                            defaultServiceIndex++;
+                    List<String> attachPoint = this.svcAttachPoints.get(typeDef.name);
+                    if (attachPoint != null) {
+                        serviceName = String.join("/", attachPoint);
```

Comment:
Don't we need to add a "/" here to the beginning ?
```suggestion
                        serviceName = "/" + String.join("/", attachPoint);
```

Also what happens when the user specify the base path as `/` or `.` ?

Reply:
Also what happens when the user specify the base path as `/` or `.` in the code ?

Reply:
The `/` prefix is not appended with `@display` annotation as well.
Do we need to add it here?

Reply:
I think we should.
Difference is that
* In `@display` annotation, we give the name as `testSvcFour`
* In basepath we give it as `/test/serviceThree`
I think we need to show the basepath that they give if we can. (check my second comment also)

Reply:
`.` is not allowed as the base path right? At the BIR level, the behaviour of `attachPoint` in `BIRServiceDeclaration` is as follows:
- If it's just `/`, it will be an empty list
- For `/foo/bar` style base path, it will return a list containing the strings `"foo"` and `"bar"`
- If a base path is not specified, `attachPoint` will be `null`.

If a literal is specified instead of a base path, `attachPoint` will be `null` and the `attachPointLiteral` will contain the literal specified.

Reply:
Then adding "/" as a prefix should fix it I guess. But then we need to handle `attachPointLiteral` literal case also in this logic.

Maybe we can generate these correctly and store in `svcAttachPoints` @madushajg  ?

Reply:
Fixed with eef9a7fa417e0d3b9f796dcdb0b31e4f7cd704f9